Sources/OTLPCore/Metrics/OTelMetricDataModel+Proto.swift

```diff
@@ -0,0 +1,209 @@
+        }
+        if let version = instrumentationScope.version {
+            self.version = version
+        }
+        attributes = .init(instrumentationScope.attributes)
+        droppedAttributesCount = UInt32(instrumentationScope.droppedAttributeCount)
+    }
+}
+
+@_spi(Metrics)
+extension Opentelemetry_Proto_Metrics_V1_Metric {
+    package init(_ metric: OTelMetricPoint) {
+        name = metric.name
+        description_p = metric.description
+        unit = metric.unit
+        switch metric.data {
+        case .gauge(let gauge):
+            self.gauge = .init(gauge)
+        case .sum(let sum):
+            self.sum = .init(sum)
+        case .histogram(let histogram):
+            self.histogram = .init(histogram)
+        }
+    }
+}
+
+@_spi(Metrics)
+extension Opentelemetry_Proto_Metrics_V1_Gauge {
+    package init(_ gauge: OTelGauge) {
+        dataPoints = .init(gauge.points)
+    }
+}
+
+@_spi(Metrics)
+extension Opentelemetry_Proto_Metrics_V1_Sum {
+    package init(_ sum: OTelSum) {
+        aggregationTemporality = .init(sum.aggregationTemporality)
+        isMonotonic = sum.monotonic
+        dataPoints = .init(sum.points)
+    }
+}
+
+@_spi(Metrics)
+extension Opentelemetry_Proto_Metrics_V1_AggregationTemporality {
+    package init(_ aggregationTemporaility: OTelAggregationTemporailty) {
+        switch aggregationTemporaility {
+        case .cumulative:
+            self = .cumulative
+        case .delta:
+            self = .delta
+        }
+    }
+}
+
+@_spi(Metrics)
+extension [Opentelemetry_Proto_Metrics_V1_NumberDataPoint] {
+    package init(_ points: [OTelNumberDataPoint]) {
+        self = points.map(Element.init)
+    }
+}
+
+@_spi(Metrics)
+extension Opentelemetry_Proto_Metrics_V1_NumberDataPoint {
+    package init(_ point: OTelNumberDataPoint) {
+        attributes = .init(point.attributes)
+        if let startTime = point.startTimeNanosecondsSinceEpoch {
+            startTimeUnixNano = startTime
+        }
+        timeUnixNano = point.timeNanosecondsSinceEpoch
+        switch point.value {
+        case .double(let value):
+            self.value = .asDouble(value)
+        case .int64(let value):
+            self.value = .asInt(value)
+        }
+        exemplars = .init(point.exemplars)
+    }
+}
+
+@_spi(Metrics)
+extension [Opentelemetry_Proto_Common_V1_KeyValue] {
+    package init(_ attributes: [OTelAttribute]) {
+        self = attributes.map(Element.init)
+    }
+}
+
+@_spi(Metrics)
+extension Opentelemetry_Proto_Common_V1_KeyValue {
+    package init(_ attribute: OTelAttribute) {
+        key = attribute.key
+        value = Opentelemetry_Proto_Common_V1_AnyValue(attribute.value)
+    }
+}
+
+@_spi(Metrics)
+extension Opentelemetry_Proto_Common_V1_AnyValue {
+    package init(_ string: String) {
+        value = .stringValue(string)
+    }
+}
+
+@_spi(Metrics)
+extension Opentelemetry_Proto_Metrics_V1_Exemplar {
+    package init(_ exemplar: OTelExemplar) {
+        // TODO:
```

Comment:
Should this TODO be resolved before merging, or do you plan to tackle it in a follow-up?

Reply:
This TODO is _not_ copy-paste from any span equivalent. Right now there's no support for exemplars and I'll add it on the end of the open questions list.